File: app/src/main/java/amrabed/android/release/evaluation/progress/ProgressSection.java
package amrabed.android.release.evaluation.progress;

import android.app.Activity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import amrabed.android.release.evaluation.R;

/**
 * Progress Section
 */
public class ProgressSection extends Fragment
{
	@Nullable
	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState)
	{
		final View view = inflater.inflate(R.layout.progress, parent, false);
		final ViewPager pager = view.findViewById(R.id.pager);

		pager.setAdapter(new PagerAdapter(getChildFragmentManager()));
		return view;
	}

	@Override
	public void onResume()
	{
		super.onResume();
		Activity activity = getActivity();
		if (activity != null) {
			activity.setTitle(R.string.menu_progress);
		}
	}

	private class PagerAdapter extends FragmentPagerAdapter {
		PagerAdapter(FragmentManager manager) {
			super(manager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
		}

		@NonNull
		@Override
		public Fragment getItem(int position) {
			return BarFragment.newInstance(position);
		}

		@Override
		public int getCount() {
			return 3;
		}

		@Override
		public CharSequence getPageTitle(int position) {
			return getResources().getStringArray(R.array.progress)[position];
		}
	}

}

File: app/src/main/java/amrabed/android/release/evaluation/main/progress/ProgressSection.kt
package amrabed.android.release.evaluation.main.progress

import amrabed.android.release.evaluation.R
import amrabed.android.release.evaluation.models.DayViewModel
import amrabed.android.release.evaluation.tools.graphs.StackedBarPlot
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import androidx.lifecycle.Observer
import androidx.viewpager2.adapter.FragmentStateAdapter
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.tabs.TabLayoutMediator
import org.joda.time.DateTime

/**
 * Progress Section
 */
class ProgressSection : Fragment() {
    override fun onCreateView(inflater: LayoutInflater, parent: ViewGroup?, state: Bundle?): View =
            inflater.inflate(R.layout.progress, parent, false).apply {
                val pager = findViewById<ViewPager2>(R.id.pager)
                pager.adapter = PagerAdapter()
                TabLayoutMediator(findViewById(R.id.title), pager) { tab, position ->
                    tab.text = resources.getStringArray(R.array.progress)[position]
                }.attach()
            }

    private inner class PagerAdapter : FragmentStateAdapter(requireActivity()) {
        override fun createFragment(position: Int) = IntervalFragment().apply { arguments = bundleOf(Pair(POSITION, position)) }
        override fun getItemCount() = 3
    }
}

/**
 * Fragment to show weekly, monthly, and yearly progress
 */
class IntervalFragment : Fragment() {
    private val viewModel by activityViewModels<DayViewModel>()

    override fun onCreateView(inflater: LayoutInflater, parent: ViewGroup?, state: Bundle?): View =
            inflater.inflate(R.layout.interval_fragment, parent, false).apply {
                val shift = nDays[requireArguments().getInt(POSITION)]
                viewModel.getRange(DateTime().minusDays(shift).millis, DateTime().millis)
                        ?.observe(viewLifecycleOwner, Observer {
                            StackedBarPlot(requireContext(), requireArguments().getInt(POSITION), it)
                                    .show(findViewById(R.id.chart))
                        })
            }
}

val nDays = intArrayOf(6, 30, 365)
const val POSITION = "position"